java/ql/src/semmle/code/java/dataflow/internal/TaintTrackingUtil.qll

```diff
@@ -680,3 +678,18 @@ private MethodAccess callReturningSameType(Expr ref) {
   ref = result.getQualifier() and
   result.getMethod().getReturnType() = ref.getType()
 }
+
+private SrcRefType entrypointType() {
+  result =
+    any(RemoteFlowSource s | s instanceof DataFlow::ExplicitParameterNode)
```

Comment:
I just played around with this a bit locally and managed a horrendous join-order by quick-eval'ing `entrypointFieldStep`, so we'll need to guard against that:
```suggestion
    pragma[only_bind_out](any(RemoteFlowSource s | s instanceof DataFlow::ExplicitParameterNode))
```